### PR TITLE
Remove root reference from file names

### PR DESCRIPTION
**Pull request by mmngreco**
Greetings!

Thank you for this project. I have found it to be useful. While using it, we
encountered an unexpected issue while reading a 7z file that contains a large
number of files and folders (I am not sure if this is relevant or not). Here is
the issue description and the solution in case this is a bug and not a result
of my incorrect usage.

Someone gave us a file which I am not able to share as it is private, and we
cannot control how it was created (it would be nice to know if there is an
option in the creation or reading process that could solve this). 

## 7z content

```sh
$ py7zr l ./eur2023_03_000-components-mn-road_el_id_mapping.7z.001.7z.001
total 167 files and directories in archive
   Date      Time    Attr         Size   Compressed  Name
------------------- ----- ------------ ------------  ------------------------
2023-02-16 08:55:20 .....     15380698     15380698  /eur2023_03_000/components/mn/road_el_id_mapping/aze/aze/azeaze_ids.csv.gz
2023-02-15 08:38:19 .....      5794553      5794553  /eur2023_03_000/components/mn/road_el_id_mapping/arm/arm/armarm_ids.csv.gz
2023-02-15 12:50:38 .....     19318146     19318146  /eur2023_03_000/components/mn/road_el_id_mapping/blr/blr/blrblr_ids.csv.gz
2023-02-16 08:40:34 .....      1893930      1893930  /eur2023_03_000/components/mn/road_el_id_mapping/mne/mne/mnemne_ids.csv.gz
2023-02-15 02:07:58 .....     25233026     25233026  /eur2023_03_000/components/mn/road_el_id_mapping/che/ch1/chech1_ids.csv.gz
2023-02-15 02:08:10 .....     12086041     12086041  /eur2023_03_000/components/mn/road_el_id_mapping/che/ch2/chech2_ids.csv.gz
2023-02-16 11:58:29 .....     11369506     11369506  /eur2023_03_000/components/mn/road_el_id_mapping/hrv/hrv/hrvhrv_ids.csv.gz
2023-02-16 10:11:56 .....      8720593      8720593  /eur2023_03_000/components/mn/road_el_id_mapping/esp/e12/espe12_ids.csv.gz
2023-02-16 10:13:11 .....     27113301     27113301  /eur2023_03_000/components/mn/road_el_id_mapping/esp/e16/espe16_ids.csv.gz
2023-02-16 10:12:41 .....     19525651     19525651  /eur2023_03_000/components/mn/road_el_id_mapping/esp/e14/espe14_ids.csv.gz
```

As you can see from the output of the `py7zr l` command, the Name starts with
`/` (root). 

```python
# Minimal Working Example
from py7zr import SevenZipFile

file = "./eur2023_03_000-components-mn-road_el_id_mapping.7z.001"
zip = SevenZipFile(file, 'r')
target = zip.getnames()[0]
seven_file = zip.read(target)
```

When we tried to read any file with the following code, we got an
error:

```python
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ <ipython-input-1-898611923896>:6 in <module>                                                     │
│                                                                                                  │
│ /home/mgreco/github/cartodb/data-observatory/etl/deps/py7zr/py7zr/py7zr.py:967 in read           │
│                                                                                                  │
│    964 │                                                                                         │
│    965 │   def read(self, targets: Optional[List[str]] = None) -> Optional[Dict[str, IO[Any]]]:  │
│    966 │   │   self._dict = {}                                                                   │
│ ❱  967 │   │   return self._extract(path=None, targets=targets, return_dict=True)                │
│    968 │                                                                                         │
│    969 │   def extract(self, path: Optional[Any] = None, targets: Optional[List[str]] = None) -  │
│    970 │   │   self._extract(path, targets, return_dict=False)                                   │
│                                                                                                  │
│ /home/mgreco/github/cartodb/data-observatory/etl/deps/py7zr/py7zr/py7zr.py:577 in _extract       │
│                                                                                                  │
│    574 │   │   │   │   │   i += 1                                                                │
│    575 │   │   │   fnames.append(outname)                                                        │
│    576 │   │   │   if path is None or path.is_absolute():                                        │
│ ❱  577 │   │   │   │   outfilename = get_sanitized_output_path(outname, path)                    │
│    578 │   │   │   else:                                                                         │
│    579 │   │   │   │   outfilename = get_sanitized_output_path(outname, pathlib.Path(os.getcwd(  │
│    580 │   │   │   if targets is not None and f.filename not in targets:                         │
│                                                                                                  │
│ /home/mgreco/github/cartodb/data-observatory/etl/deps/py7zr/py7zr/helpers.py:478 in              │
│ get_sanitized_output_path                                                                        │
│                                                                                                  │
│   475 │   │   outfile = canonical_path(path.joinpath(remove_relative_path_marker(fname)))        │
│   476 │   │   if is_relative_to(outfile, path):                                                  │
│   477 │   │   │   return pathlib.Path(outfile)                                                   │
│ ❱ 478 │   raise Bad7zFile(f"Specified path is bad: {fname}")                                     │
│   479                                                                                            │
│   480                                                                                            │
│   481 def check_archive_path(arcname: str) -> bool:                                              │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
Bad7zFile: Specified path is bad: /eur2023_03_000/components/mn/road_el_id_mapping/aze/aze/azeaze_ids.csv.gz
```

I had the intuition that the problem was related to the root reference at the
beginning of the file name, but I couldn't find any documentation about it. So,
here is my solution (very naive/silly, but it works).

In addition, looking at the code, I was wondering why not use `Path` instead of
`PurePath`? With `Path`, you can use `.resolve()`, `.expanduser()`,
`.absolute()`, and many other methods that may replace (I think) functions
entirely (like `canonical_path`). But this probably should be placed in another
issue.

### CODE REVIEW

**Comment by miurahr** [comment]
Thank you for the proposal, @mmngreco . Could you add small regression test for the case?

In my understanding, when  using ``Path`` , it will access actuall filesystem, and can be failed by file system error. 
A path in archive should not be access to actualll path except for extracting or archiveing. It is why I use ``PurePath`` instead of ``Path`` for internal path handling that come from archive meta data.

**Comment by mmngreco** [comment]
> Thank you for the proposal, @mmngreco . Could you add small regression test for the case?

Sure! could you give me some guidance about it? In which test file should I add the test? there's a convention that I should take into account?

**Comment by miurahr** [comment]
It seems a case of extract,  a test would be added into test_extract.py
A file should be stored in tests/data/ folder with a name of test, or character of a contents.

**Comment by miurahr** [comment]
Test failed with coding style of test code.
Could you please try running ``black`` code formatter command?

File: tests/test_extract.py [hunk 1]
```diff
@@ -581,3 +581,29 @@ def test_extract_hidden_linux_file(tmp_path):
...
+
+@pytest.mark.files
+def test_extract_root_path_arcname(tmp_path):
+    # This test checks the extraction of a file with a arcname like
+    # "/a/b/test.txt" which shouldn't fail.
+    filename_7z = testdata_path.joinpath("root_path_arcname.7z")
+    content = bytes("This is a test", "ascii")
+    filename = "a/b/test.txt"  # expected arcname
+
+    with py7zr.SevenZipFile(filename_7z, 'r') as archive:
```
Comment (miurahr) [inline tests/test_extract.py:594]:
```suggestion
    with py7zr.SevenZipFile(filename_7z, "r") as archive:
```

**Comment by mmngreco** [comment]
Do you need something else to get this merged?

**Pull request merged**